utilities/oadp.py

```diff
@@ -0,0 +1,173 @@
+
+    def __exit__(self, exception_type, exception_value, traceback):
+        try:
+            if getattr(self, "teardown", True):
+                delete_velero_resource(resource=self, client=self.client)
+            else:
+                LOGGER.info("Skipping Velero delete for %s %s (teardown=False)", self.kind, self.name)
+        except Exception:
+            LOGGER.exception("Failed to delete Velero %s %s", self.kind, self.name)
+        finally:
+            super().__exit__(exception_type, exception_value, traceback)
+
+
+@contextmanager
+def create_rhel_vm(
+    storage_class,
+    namespace,
+    dv_name,
+    vm_name,
+    rhel_image,
+    client=None,
+    wait_running=True,
+    volume_mode=None,
+):
+    artifactory_secret = None
+    artifactory_config_map = None
+
+    try:
+        artifactory_secret = get_artifactory_secret(namespace=namespace)
+        artifactory_config_map = get_artifactory_config_map(namespace=namespace)
+
+        dv = DataVolume(
+            name=dv_name,
+            namespace=namespace,
+            source="http",
+            url=get_http_image_url(
+                image_directory=Images.Rhel.DIR,
+                image_name=rhel_image,
+            ),
+            storage_class=storage_class,
+            size=Images.Rhel.DEFAULT_DV_SIZE,
+            api_name="storage",
+            volume_mode=volume_mode,
+            secret=artifactory_secret,
+            cert_configmap=artifactory_config_map.name,
+        )
+        dv.to_dict()
+        dv_metadata = dv.res["metadata"]
+        with VirtualMachineForTests(
+            client=client,
+            name=vm_name,
+            namespace=dv_metadata["namespace"],
+            os_flavor=OS_FLAVOR_RHEL,
+            memory_guest=Images.Rhel.DEFAULT_MEMORY_SIZE,
+            data_volume_template={"metadata": dv_metadata, "spec": dv.res["spec"]},
+            run_strategy=VirtualMachine.RunStrategy.ALWAYS,
+        ) as vm:
+            if wait_running:
+                running_vm(vm=vm, wait_for_interfaces=True)
+            yield vm
+    finally:
+        cleanup_artifactory_secret_and_config_map(
+            artifactory_secret=artifactory_secret, artifactory_config_map=artifactory_config_map
+        )
+
+
+def check_file_in_vm(vm):
+    with console.Console(vm=vm) as vm_console:
+        vm_console.sendline(LS_COMMAND)
+        vm_console.expect(FILE_NAME_FOR_BACKUP, timeout=TIMEOUT_20SEC)
+        vm_console.sendline(f"cat {FILE_NAME_FOR_BACKUP}")
+        vm_console.expect(TEXT_TO_TEST, timeout=TIMEOUT_20SEC)
+
+
+def is_storage_class_support_volume_mode(storage_class_name, requested_volume_mode):
+    """
+    Check if the storage class supports the requested volume mode.
+    """
+    # If no specific mode is requested or Filesystem is requested, assume supported.
+    if not requested_volume_mode:
+        return True
+    if isinstance(requested_volume_mode, str) and requested_volume_mode.lower() != "block":
+        return True
```

Comment:
- These statements should be combined into one.
- Instead of checking for str/lower(), can we use DataVolume.VolumeMode.BLOCK?